src/lib/services/client-metrics/metrics-service-v2.ts

```diff
@@ -229,4 +229,11 @@ export default class ClientMetricsServiceV2 {
         }
         return 'default';
     }
+
+    resolveUserEnvironment(user: IUser | IApiUser): string {
```

Comment:
Can we post metrics as a regular user?

Reply:
No, you need a Server Token.

Reply:
Yes, not that Typescript will care much about, but at least we'd be more clear.

Reply:
Made it clearer and it only accepts IApiUser now.

Reply:
Hmm, I'm going to keep the `IUser | IApiUser` here, we don't have a good setup for testing with custom api tokens at the moment, and this allows the test to verify that we do filter by environment token has access to, since if we don't have an API token we can only have access to 'default'